ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/SeedService.scala

```diff
@@ -33,38 +34,51 @@ object SeedService {
     seeding match {
       case Seeding.Strong => StrongRandom
       case Seeding.Weak => WeakRandom
-      case Seeding.Static => new StaticRandom("static random seed service")
+      case Seeding.Static => staticRandom("static random seed service")
     }
 
-  // Pseudo random generator seeded with high entropy seed.
-  // May block while gathering entropy from the underlying operating system
-  // Thread safe.
-  object StrongRandom extends SeedService {
-    override val nextSeed: () => crypto.Hash =
-      crypto.Hash.secureRandom(
-        crypto.Hash.assertFromByteArray(
-          SecureRandom.getInstanceStrong.generateSeed(crypto.Hash.underlyingHashLength)
-        )
-      )
+  /** Pseudo random generator seeded with high entropy seed.
+    * May block while gathering entropy from the underlying operating system
+    * Thread safe.
+    */
+  // lazy to avoid gathering unnecessary entropy.
+  lazy val StrongRandom = {
+    val logger = LoggerFactory.getLogger(this.getClass)
+    var done = false
```

Comment:
```suggestion
    @volatile var done = false
```
Again, probably fine, but if you use mutable state and threads this is possibly sensible to have.